packages/airnode-adapter/src/response-processing/extraction.ts

```diff
@@ -139,5 +143,11 @@ export function extractAndEncodeResponse(data: unknown, parameters: ReservedPara
   }
 
   const extractedValue = extractSingleResponse(data, parameters);
-  return { rawValue: data, encodedValue: encodeValue(extractedValue, parameters._type), values: [extractedValue] };
+  const encodedValue = encodeValue(extractedValue, parameters._type);
+
+  if (encodedValue.length > MAX_ENCODED_RESPONSE_SIZE) {
+    throw new Error(`Encoded value exceeds the maximum allowed size (${MAX_ENCODED_RESPONSE_SIZE} bytes)`);
+  }
```

Comment:
It looks like you're checking the stringified length here. I recommend using `ethers.utils.arrayify` and checking the length of its output to get the length in bytes

Reply:
Yeah, I was wondering whether comparing the bytesString length is actually correct.